### PR TITLE
JS: Added support for numeric separators

### LINKED ISSUE #1893: JavaScript numeric separators
**Information**
- Language: JavaScript
- Plugins: none

Does the problem still occur in the latest version of Prism? You can check using the [test page](https://prismjs.com/test.html) or get the latest version at the [download page](https://prismjs.com/download.html).

Yes

**Description**
A clear and concise description of what is being highlighted incorrectly and how it should be highlighted instead. Add screenshots to illustrate the problem.

The use of [numeric separators](https://github.com/tc39/proposal-numeric-separator) causes numeric literals to be highlighted incorrectly.

<details>
<summary>The code being highlighted incorrectly.</summary>

```
1_000_000_000_000
    1_019_436_871.42
```

</details>

Screenshot from <https://v8.dev/blog/v8-release-75#numeric-separators>:

### PR DESCRIPTION
**Pull request by RunDevelopment**
This fixes #1893.

The `number` pattern is quite a bit longer because of this change, so any ideas on how we could simplify this are more than welcome.

---

[Spec changes](https://github.com/tc39/proposal-numeric-separator/blob/master/spec.md#syntax).

### CODE REVIEW

File: tests/languages/javascript/number_feature.test [hunk 1]
```diff
@@ -7,11 +7,21 @@
...
 NaN
 Infinity
+
 123n
 0x123n
 
+1_000_000_000_000
+1_000_000.220_720
+0b0101_0110_0011_1000
+0x40_76_38_6A_73
```
Comment (mathiasbynens) [inline tests/languages/javascript/number_feature.test]: Could add a test for `0o12_34_56` for completeness’ sake (although I don’t think there’s a good use case for separators in such literals)

Comment (mathiasbynens) [reply tests/languages/javascript/number_feature.test]: https://github.com/tc39/proposal-numeric-separator/issues/44

Comment (RunDevelopment) [reply tests/languages/javascript/number_feature.test]: Added.

**Comment by RunDevelopment** [comment]
@mathiasbynens Thank you for reviewing!

**Comment by mathiasbynens** [comment]
@RunDevelopment Thanks for the quick patch! 

There’s no rush, but what’s the general process for releasing a new `prism-js` version on npm? Is there a fixed release schedule or something like that?

**Comment by RunDevelopment** [comment]
There is no fixed release schedule. A new version usually gets releases once enough changes have been made.
Looking at the committed changes since 1.16 and the pending PRs, I would say that we will release 1.17 soon (maybe in a few weeks or so).

**Pull request merged**